migration/migration.go

```diff
@@ -1188,4 +1190,413 @@ func removeInvalidUserPreferences(usersPrefs []v1schema.UserPreferences) (update
+	}
+
+	if err = cursor.Err(); err != nil {
+		reqLogger.Error(err, "MongoDB cursor encountered an error")
+		return
+	}
+	reqLogger.Info("Successfully copied over ScimServerUsers to EDB", "docsMigrated", migrateCount)
+	return
+}
+
+func getSSGroupFromMap(result map[string]any) (ssGroup *v1schema.ScimServerGroup, err error) {
+	ssGroup = &v1schema.ScimServerGroup{}
+	if err = v1schema.ConvertToScimServerGroup(result, ssGroup); err != nil {
+		err = fmt.Errorf("failed to unmarshal into scimservergroup: %w", err)
+		ssGroup = nil
+		return
+	}
+	return
+}
+
+func insertSSGroup(ctx context.Context, postgres *PostgresDB, ssGroup *v1schema.ScimServerGroup) (err error) {
+	reqLogger := logf.FromContext(ctx)
+	args := ssGroup.GetArgs()
+	query := ssGroup.GetInsertSQL()
+	var id *string
+	err = postgres.Conn.QueryRow(ctx, query, args).Scan(&id)
+	if errors.Is(err, pgx.ErrNoRows) {
+		reqLogger.Info("Row already exists in EDB table platformdb.scim_server_groups")
+		return nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to INSERT into table", "table", "platformdb.scim_server_groups")
+		return err
+	}
+	return
+}
+
+func insertSSGroupCustom(ctx context.Context, postgres *PostgresDB, ssGroupCustom *v1schema.ScimServerGroupCustom) (err error) {
+	reqLogger := logf.FromContext(ctx)
+	args := ssGroupCustom.GetArgs()
+	query := ssGroupCustom.GetInsertSQL()
+	var id *string
+	err = postgres.Conn.QueryRow(ctx, query, args).Scan(&id)
+	if errors.Is(err, pgx.ErrNoRows) {
+		reqLogger.Info("Row already exists in EDB table platformdb.scim_server_groups_custom")
+		return nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to INSERT into table", "table", "platformdb.scim_server_groups_custom")
+		return err
+	}
+	return
+}
+
+func insertSSGroupRelatedRows(ctx context.Context, mongodb *MongoDB, postgres *PostgresDB) (err error) {
+	dbName := "platform-db"
+	collectionName := "ScimServerGroups"
+	reqLogger := logf.FromContext(ctx)
+	filter := bson.D{{Key: "migrated", Value: bson.D{{Key: "$ne", Value: true}}}}
+	cursor, err := mongodb.Client.Database(dbName).Collection(collectionName).Find(ctx, filter)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get cursor from MongoDB")
+		return
+	}
+	errCount := 0
+	migrateCount := 0
+	for cursor.Next(ctx) {
+		var ssGroup *v1schema.ScimServerGroup
+		// mongo doc is split into 3 maps
+		// ssGroupMap - goes as a row into scim_server_groups table
+		// ssGroupCustomMap - goes as a multiple rows into scim_server_groups_custom table
+		// ssGroupCustomSchemaMap - goes as a multiple rows into scim_server_groups_custom table
+		var ssGroupMap map[string]interface{}
+		if err = cursor.Decode(&ssGroupMap); err != nil {
+			reqLogger.Error(err, "Failed to decode Mongo document")
+			errCount++
+			err = nil
+			continue
+		}
+		ssGroupCustomMap := populateSSGroupCustomMap(ssGroupMap)
+		ssGroupCustomSchemaMap := populateSSGroupCustomSchemaMap(ssGroupCustomMap)
+		ssGroup, err = getSSGroupFromMap(ssGroupMap)
```

Comment:
Same comments as above re: handling this error.

Reply:
Addressed